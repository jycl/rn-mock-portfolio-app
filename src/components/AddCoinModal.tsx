import React from 'react';
import {
  Dimensions,
  View,
  Button,
  Modal,
  Text,
  StyleSheet,
} from 'react-native';
import { TextInput } from 'react-native-gesture-handler';

const screenWidth = Dimensions.get('window').width;

type PropsType = {
  onDismiss?: () => void;
};

/**
 * Modal to show users a list of cryptocurrencies to search through
 * and add to existing list
 */

export default function AddCoinModal({ onDismiss }: PropsType) {
  return (
    <Modal
      presentationStyle="formSheet"
      animationType="slide"
      onDismiss={onDismiss}
      transparent={false}
      onRequestClose={onDismiss}>
      <View style={styles.container}>
        <Button title="Back" onPress={() => onDismiss && onDismiss()} />
        <View style={styles.innerContainer}>
          <Text style={styles.text}>Please select a coin to add to list:</Text>
          <TextInput
            style={styles.textInput}
            placeholder="Enter cryptocurrency name..."
            placeholderTextColor="#808080"
          />
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#1c1c1c',
    padding: 12,
    alignItems: 'flex-start',
  },
  innerContainer: {
    margin: 8,
  },
  text: {
    color: 'white',
    fontSize: 18,
    fontWeight: 'bold',
  },
  textInput: {
    color: '#000',
    borderRadius: 8,
    padding: 4,
    height: 40,
    width: screenWidth * 0.9,
    backgroundColor: '#FFF',
    marginTop: 12,
    fontSize: 18,
  },
});
