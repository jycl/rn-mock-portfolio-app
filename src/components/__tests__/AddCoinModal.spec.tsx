import React from 'react';
import TestRenderer from 'react-test-renderer';
import AddCoinModal from '../AddCoinModal';

describe('AddCoinModal component', () => {
  it('should match snapshot rendering given defined props', () => {
    const renderItem = TestRenderer.create(<AddCoinModal />);
    expect(renderItem.toJSON()).toMatchSnapshot();
  });
});
