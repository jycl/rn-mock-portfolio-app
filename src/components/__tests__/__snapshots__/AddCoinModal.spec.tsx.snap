// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCoinModal component should match snapshot rendering given defined props 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  presentationStyle="formSheet"
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#1c1c1c",
        "flex": 1,
        "padding": 12,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Please select a coin to add to list:
      </Text>
      <TextInput
        allowFontScaling={true}
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        placeholder="Enter cryptocurrency name..."
        placeholderTextColor="#808080"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 8,
            "color": "#000",
            "fontSize": 18,
            "height": 40,
            "marginTop": 12,
            "padding": 4,
            "width": 675,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</Modal>
`;
